### ISSUE TITLE
`invalid-paramspec` example from docs does not report the error

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ty#3734]
https://docs.astral.sh/ty/reference/rules/#invalid-paramspec

it reports `mismatched-type-name` instead

```py
from typing import ParamSpec

P1 = ParamSpec("P1")  # okay
P2 = ParamSpec("S2")  # error: mismatched-type-name
```

[Playground](https://play.ty.dev/dba177fa-536a-4ae8-bb44-f8e22a1a9ea3)
